docs/API/API-functions.md

```diff
@@ -358,27 +358,84 @@ int srt_bind(SRTSOCKET u, const struct sockaddr* name, int namelen);
+to which they expect connection from the peer. Let's say, we have a Party 1
+that selects an endpoint A and a Party 2 that selects an endpoint B. In this case the Party 1
+binds the socket to the endpoint A and then connects to the endpoint B, and the Party 2
+the other way around. Both sockets must be set
+[`SRTO_RENDEZVOUS`](API-socket-options.md#SRTO_RENDEZVOUS) to *true* to make
+this connection possible.
+
+For a connecting socket the call to `srt_bind` is optional, but can be used to set up the
+outgoing port for communication as well as the local interface through which
+it should reach out to the remote endpoint, should that be necessary.
+
+Whether binding is possible depends on some runtime conditions, in particular:
+
+* No socket in the system has been bound to this port ("free binding"), or
+
+* A socket bound to this port is bound to a certain address, and this binding is
+  using a different non-wildcard address ("side binding"), or
+
+* A socket bound to this port is bound to a wildcard address for a different IP
+  version than the version requested for this binding ("side wildcard binding",
+  see also `SRTO_IPV6ONLY` socket option).
+
+It is also possible to bind to the already busy port as long as the existing
+binding ("shared binding") is possessed by an SRT socket created in the same
+application, and:
+
+* Its binding address and UDP-related socket options match the socket to be bound.
+* Its [`SRTO_REUSEADDR`](API-socket-options.md#SRTO_REUSEADDRS) is set to *true* (default).
+
+If none of the free, side and shared binding options is currently possible, this function
+will fail. If the socket blocking the requested endpoint is an SRT
+socket in the current application, it will report the `SRT_EBINDCONFLICT` error,
+while if it was another socket in the system, or the problem was in the system
+in general, it will report `SRT_ESOCKFAIL`. Here is the table that shows possible situations:
+
+| Address in srt_bind | Result for attempted binding |           |                             |               |               |
+|---------------------|------------------------------|-----------|-----------------------------|---------------|---------------|
+| Bind address        | A.B.C.D                      | 0.0.0.0   | ::X                         | :: / V6ONLY=1 | :: / V6ONLY=0 |
+| 1.2.3.4             | 1.2.3.4 shareable, else free | blocked   | free                        | free          | blocked       |
+| 0.0.0.0             | blocked                      | shareable | free                        | free          | blocked       |
+| 8080::1             | free                         | free      | 8080::1 sharable, else free | blocked       | blocked       |
+| :: / V6ONLY=1       | free                         | free      | blocked                     | sharable      | blocked       |
+| :: / V6ONLY=0       | blocked                      | blocked   | blocked                     | blocked       | sharable      |
+
```

Comment:
I think some explanation may be also needed

```suggestion
Where:

* free: this binding can coexist with the requested binding

* blocked: this binding conflicts with the requested binding

* shareable: this binding can be shared with the requested binding, if it's compatible

* (ADR) shareable, else free: this binding is shareable, if the requested binding address is
equal to the given one, otherwise it's free

If the binding is shareable, then the operation will succeed, if the socket that currently
occupies the binding has the `SRTO_REUSEADDR` option set to true (default) and all UDP
settings are the same as in the current socket, otherwise it will fail. Shared binding means
sharing the underlying UDP socket and communication queues between SRT sockets.
```

Reply:
Some very minor edits. What does "ADR" stand for?

```suggestion
Where:
* free: This binding can coexist with the requested binding.
* blocked: This binding conflicts with the requested binding.
* shareable: This binding can be shared with the requested binding if it's compatible.
* (ADR) shareable, else free: this binding is shareable if the requested binding address is
equal to the given one. Otherwise it's free.
If the binding is shareable, then the operation will succeed if the socket that currently
occupies the binding has the `SRTO_REUSEADDR` option set to true (default) and all UDP
settings are the same as in the current socket. Otherwise it will fail. Shared binding means
sharing the underlying UDP socket and communication queues between SRT sockets. If
all existing bindings on the same port are "free" then the requested binding will
allocate a distinct UDP socket for this SRT socket ("side binding").

```